### PR TITLE
Minor clean up of datetime and other classes

### PR DESCRIPTION
**Pull request by MitchellGale-BitQuill**
### Description
Clean up lexer to make future changes easier. Make changes to documentation and code to make cleaner. 

* Groom ANTLR grammar files to make future changes easier.
* Fix `EqualsAndHashCode` compilation warnings
>
```
Generating equals/hashCode implementation but without a call to superclass, even though this class does not extend java.lang.Object. If this is intentional, add '@EqualsAndHashCode(callSuper=false)' to your type.
```
* Move `ExprTimestampValue.ZONE` constant to `DateTimeUtils`
* Move `ExprDatetimeValue` formatter to `DateTimeFormatters`
* Sort alphabetically functions in `DateTimeFunction` and `MathematicalFunction`
* Replace `FunctionDSL` import by static imports to simplify code
* Remove tests for `NULL` and `MISSING` values
* Add validation for non-zero divider
* Remove `Environment` mock from some unit tests, because it is not needed anymore
* Groom docs to eliminate `rst` validator warnings
* Add doctests for some functions
* Make IT for `now`-like functions truly parametrized
* Rename some PPL grammar entities to be consistent with SQL ones

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: core/src/main/java/org/opensearch/sql/utils/DateTimeUtils.java
Comment (GabeFernandez310) [inline core/src/main/java/org/opensearch/sql/utils/DateTimeUtils.java:155]: Should constants be at the beginning or at the end?

File: core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java
Comment (GabeFernandez310) [inline core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java:209]: I'm curious on what these removed asserts actually do. I noticed them as I've been working on DateTime stuff, but I don't actually know what they are supposed to be testing.

File: core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java
Comment (GabeFernandez310) [inline core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java:447]: These will probably make my [PR 204](https://github.com/Bit-Quill/opensearch-project-sql/pull/204/files#diff-bdc6fcaa42a4347bba30e99a1f4434587434f56a26941a3b1f2cf6eed4ddcf0a) for `time_format` conflict. Is it safe to checkout this file into my PR, or should I wait until this is merged upstream and rebase before making my PR for `time_format` upstream.

File: docs/user/ppl/functions/datetime.rst
Comment (GabeFernandez310) [inline docs/user/ppl/functions/datetime.rst:512]: I have also changed this in my [PR](https://github.com/Bit-Quill/opensearch-project-sql/pull/204/files#diff-d77a6d15ba9b05d3242ae66e7b487ceebcf8c5fd8ee8eb5c1eb2b9e337d59d7c). Should I revert my changes?

File: opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/RelevanceQuery.java
Comment (GabeFernandez310) [reply opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/RelevanceQuery.java:92]: Sorry ignore this. Just built and ran the query and it still works.

**Review by GabeFernandez310 — APPROVED** [review]

File: docs/user/ppl/functions/datetime.rst
Comment (Yury-Fridlyand) [reply docs/user/ppl/functions/datetime.rst:512]: No!
That would be interpreted as merge conflict, which is easy to resolve.

File: opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/RelevanceQuery.java
Comment (Yury-Fridlyand) [reply opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/RelevanceQuery.java:92]: There was a duplicate function. `@Getter` generates the same.

File: core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java
Comment (Yury-Fridlyand) [reply core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java:209]: They handle `nullMissingHandling`, which is tested in `FunctionDSLnullMissingHandlingTest`. Thanks Max by adding those tests.

This specific line mocks type of `nullRef` to let repository pick a function signature.
The null/missing validation actually doesn't test the function and it is not needed to repeat it 100500 times for each function.

File: core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java
Comment (Yury-Fridlyand) [reply core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java:447]: It doesn't matter which PR goes first, but we have to rebase second one once first merged and fix conflicts.

File: core/src/main/java/org/opensearch/sql/ast/statement/Explain.java
Comment (acarbonetto) [inline core/src/main/java/org/opensearch/sql/ast/statement/Explain.java:19]: 🎖️

File: core/src/main/java/org/opensearch/sql/data/model/ExprTimestampValue.java
Comment (acarbonetto) [inline core/src/main/java/org/opensearch/sql/data/model/ExprTimestampValue.java:39]: Is there any interest in including a default time zone in the functionalProperties object?

File: core/src/main/java/org/opensearch/sql/utils/DateTimeUtils.java
Comment (acarbonetto) [reply core/src/main/java/org/opensearch/sql/utils/DateTimeUtils.java:155]: Do we also want to "encourage" private functions to be defined after public functions?

**Review by acarbonetto — APPROVED** [review]

**Review by GumpacG — APPROVED** [review]

**Review by matthewryanwells — APPROVED** [review]

**Pull request merged**
